NetCoreRestApi/DataLayer/Repositories/CategoryRepository.cs

```diff
@@ -4,32 +4,29 @@
 
 namespace DataLayer.Repositories
 {
-    public class CategoryRepository : IRepository<CategoryModel>
+    public class CategoryRepository<TId> : ICategoryRepository<TId>
```

Comment:
Remove TId, Create generic BaseRepository<TModel, TEntity, Tid> and fix class to CategoryRepository : BaseRepository<CategoryModel, CategoryEntity, int>, ICategoryRepository

Reply:
done